### DISCUSSION TITLE
How to stop unprompted requests to metadata providers?

### DISCUSSION THREAD

**Discussion by GiFuJo** [advplyr/audiobookshelf#3578] [Q&A] [answered]
When editing a library entry it seems that a request to the last selected metadata provider is made automatically just by clicking the "match" tab and without using the search button. I include some personal writing projects in audiobookshelf and I feels weird sending my name and a work title to the last selected metadata provider just like that. Are there other parts of the app that do that? Can I somehow change this behaviour? I created a custom metadata provider that just points to localhost, but that's a really annoying bandaid fix.

**Comment by mikiher** [comment]
Why would you click the `Match` tab if you don't want to match the book?

I believe that for most users, searching the default metadata provider with the book's title and author is just what they expect to happen when they click on the `Match` tab. What is the behavior the you'd expect instead? waiting until the user clicks on the `Search` button?
That is of course possible, but is that the right behavior for most people most of the times?

Also, note that the server goes to great lengths to find a good match based on whatever is in author and title fields, only to save users from editing them and searching again. The guiding principle is to minimize the amount of interaction required for finding a match.

I'd be interested to hear what other people think.

**Reply by GiFuJo** [reply]
I realize that adding an additional click here would be not optimal.

Maybe change the logic of the metadata provider selection in the match tab? Currently it uses the last selected metadata provider regardless of the selected library. Maybe instead use the default metadata provider set in the library?

**Reply by pwinnski** [reply, marked as answer]
I recognize that your situation isn't ideal, but I can't think of any solution for you that doesn't make things worse for everyone who is not you. Even using the default metadata provider rather than the last-used means breaking the existing workflow for people who want to, for example, match a bunch of UK books in a row, despite their normal provider being US-based.

I think maybe you might avoid clicking the Match tab when your personal items are selected, or if you're staying on the match tab and stepping through items, maybe put your own personal items in a different library. Or since you mentioned creating a nerfed matcher, just remember to always match against that after every time you match using a real provider, so it's still the one used when you attempt to match your personal items.

**Reply by GiFuJo** [reply]
Yeah, that's a solid argument, so I'll mark this as the answer.

For people who may have the same problem as me: I now use the following (llm co-written) script to replace the default click event listener on the "Match" tab with a double click event listener. This prevents me from accidentally clicking the "Match" tab.

```JavaScript
(function() {
    'use strict';

    function modifyMatchTab() {
        const matchTab = Array.from(document.querySelectorAll('div'))
            .find(el => el.textContent.trim() === "Match");

        if (matchTab && !matchTab.hasAttribute('data-dblclick-added')) {
            matchTab.setAttribute('data-dblclick-added', 'true');

            let doubleClickTriggered = false;

            matchTab.addEventListener('click', function(event) {
                if (!doubleClickTriggered) {
                    event.stopImmediatePropagation();
                    event.preventDefault();
                }
                doubleClickTriggered = false;
            }, true);

            matchTab.addEventListener('dblclick', function(event) {
                event.preventDefault();
                event.stopPropagation();

                window.getSelection().removeAllRanges();

                doubleClickTriggered = true;

                const mouseEvent = new MouseEvent('click', {
                    bubbles: true,
                    cancelable: true,
                    view: window
                });
                matchTab.dispatchEvent(mouseEvent);
            });

        }
    }

    const observer = new MutationObserver(() => {
        modifyMatchTab();
    });

    observer.observe(document.body, { childList: true, subtree: true });

    modifyMatchTab();
})();
```
